src/Components/Blueprints/BuildImagesButton.tsx

```diff
@@ -1,20 +1,160 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-import { Button, ButtonProps } from '@patternfly/react-core';
+import {
+  Dropdown,
+  MenuToggle,
+  MenuToggleElement,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  Flex,
+  FlexItem,
+  Spinner,
+  MenuToggleAction,
+  ButtonProps,
+  Button,
+} from '@patternfly/react-core';
+import { skipToken } from '@reduxjs/toolkit/query';
 
 import { selectSelectedBlueprintId } from '../../store/BlueprintSlice';
 import { useAppSelector } from '../../store/hooks';
-import { useComposeBlueprintMutation } from '../../store/imageBuilderApi';
+import {
+  ImageTypes,
+  useComposeBlueprintMutation,
+  useGetBlueprintQuery,
+} from '../../store/imageBuilderApi';
 
-type BuildImagesButtonPropTypes = {
+export const BuildImagesButton = () => {
+  const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
+  const [selectedTargets, setSelectedTargets] = useState<ImageTypes[]>([]);
+  const [buildBlueprint, { isLoading: imageBuildLoading }] =
+    useComposeBlueprintMutation();
+  const onBuildHandler = async () => {
+    if (selectedBlueprintId) {
+      await buildBlueprint({
```

Comment:
I'm not sure here, but where do we handle errors? :thinking:
@regexowl / @amirfefer ?

Reply:
That's the neat part - we don't... but in reality we probably should. It would be similar to the loading logic, what needs to be done is adding error to the mutation:

```typescript
  const [buildBlueprint, { isLoading: imageBuildLoading, isError: imageBuildError }] =
    useComposeBlueprintMutation();
```

And then do something like this probably?

```typescript
  try {
    await buildBlueprint({
      ...
  catch (imageBuildError) {
    dispatch(
      addNotification({
        <something's wrong>
```

Seems like we seldom properly handle errors in mutations though. :face_with_peeking_eye: